polaris/polaris/tests/auth_test.py

```diff
@@ -43,7 +43,7 @@ def test_auth_get_account(client):
 
     manage_data_op = transaction_object.operations[0]
     assert manage_data_op.type_code() == Xdr.const.MANAGE_DATA
-    assert manage_data_op.data_name == f"{urlparse(settings.HOST_URL).hostname} auth"
```

Comment:
This test failed when `polaris/.env`'s `HOST_URL` param was `HOST_URL="https://localhost:8000"` as suggested in the documentation https://django-polaris.readthedocs.io/en/stable/tutorials/index.html?highlight=env#specify-environment-variables

However it passed all tests when I removed the port number.
If we want to remain consistent I think we should either change how this pulls the the hostname back to its original behavior
or 
change the documentation to communicate that the port number in `polaris/.env`'s `HOST_URL` is not needed

Reply:
Thats strange, are you sure? I think its the other way around:

`netloc` includes the port number, so when `HOST_URL=http://localhost:8000` it should match, because `SEP10_HOME_DOMAINS` defaults to `urlparse(settings.HOST_URL).netloc`.

`hostname` doesn't have the port number, so when `HOST_URL=http://localhost:8000` is should fail because:
```
>>> urlparse(settings.HOST_URL).hostname
localhost
```
And the polaris settings have it as:
```
>>> urlparse(settings.HOST_URL).netloc
localhost:8000
```

Reply:
I'm pretty sure.
I got this error when my HOST_URL was "https://localhost:8000"

```
=================================================================== FAILURES ===================================================================
____________________________________________________________ test_auth_get_account _____________________________________________________________

client = <django.test.client.Client object at 0x7f9751115250>

    def test_auth_get_account(client):
        """`GET <auth>` succeeds with a valid TransactionEnvelope XDR."""
        response = client.get(f"{endpoint}?account={STELLAR_ACCOUNT_1}", follow=True)
        content = json.loads(response.content)
        assert content["network_passphrase"] == "Test SDF Network ; September 2015"
        assert content["transaction"]
    
        envelope_xdr = content["transaction"]
        envelope_object = TransactionEnvelope.from_xdr(
            envelope_xdr, network_passphrase=settings.STELLAR_NETWORK_PASSPHRASE
        )
        transaction_object = envelope_object.transaction
        assert transaction_object.sequence == 0
        assert len(transaction_object.operations) == 1
    
        manage_data_op = transaction_object.operations[0]
        assert manage_data_op.type_code() == Xdr.const.MANAGE_DATA
>       assert manage_data_op.data_name == f"{urlparse(settings.HOST_URL).netloc} auth"
E       AssertionError: assert 'localhost auth' == 'localhost:8000 auth'
E         - localhost:8000 auth
E         ?          -----
E         + localhost auth

polaris/polaris/tests/auth_test.py:46: AssertionError
------------------------------------------------------------- Captured stderr call -------------------------------------------------------------
2020-11-25 16:29:33,988 - INFO - polaris.sep10.views: Returning SEP-10 challenge for account GAIRMDK7VDAXKXCX54UQ7WQUXZVITPBBYH33ADXQIADMDTDVJMQGBQ6V
-------------------------------------------------------------- Captured log call ---------------------------------------------------------------
INFO     polaris.sep10.views:views.py:72 polaris.sep10.views: Returning SEP-10 challenge for account GAIRMDK7VDAXKXCX54UQ7WQUXZVITPBBYH33ADXQIADMDTDVJMQGBQ6V
=============================================================== warnings summary ===============================================================
../../.virtualenvs/django-polaris-vKUchHve/lib/python3.7/site-packages/urllib3/util/retry.py:255
  /Users/lawrence/.virtualenvs/django-polaris-vKUchHve/lib/python3.7/site-packages/urllib3/util/retry.py:255: DeprecationWarning: Using 'method_whitelist' with Retry is deprecated and will be removed in v2.0. Use 'allowed_methods' instead
    DeprecationWarning,

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================================================== short test summary info ============================================================
FAILED polaris/polaris/tests/auth_test.py::test_auth_get_account - AssertionError: assert 'localhost auth' == 'localhost:8000 auth'
```

Reply:
This branch has the following definition in `polaris/polaris/settings.py`
```
SEP10_HOME_DOMAINS = env_or_settings("SEP10_HOME_DOMAINS", list=True) or [
    urlparse(HOST_URL).netloc
]
```
So it doesn't make sense that it would fail to match:
```
f"{urlparse(settings.HOST_URL).netloc} auth"
```

Reply:
Tests are passing in this PR, why is your branch different?

Reply:
If I'm understanding this correctly youre referring to this piece of code.
```
HOST_URL = env_or_settings("HOST_URL")
if not HOST_URL.startswith("http"):
    raise ImproperlyConfigured("HOST_URL must include a protocol (http or https)")

SEP10_HOME_DOMAINS = env_or_settings("SEP10_HOME_DOMAINS", list=True) or [
    urlparse(HOST_URL).netloc
]
```
Then that means `f"{urlparse(settings.HOST_URL).netloc} auth"` is looking at `env_or_settings("HOST_URL")` correct?

Reply:
If the intention was to pull the `SEP10_HOME_DOMAINS` then we would need to change this to `f"{urlparse(settings.SEP10_HOME_DOMAINS).netloc} auth"` but that poses different problems if its a list of home domains if Im not mistaken.

Reply:
You could do something like this if we were looking at the `SEP10_HOME_DOMAINS`
But its very inelegant 
`f"{urlparse(settings.SEP10_HOME_DOMAINS[0]).path} auth"`